workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/hooks/useHash.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useCallback, useEffect, useState} from 'react';
+
+export function useHash(defaultHash: string) {
+	const [hash, setHash] = useState(() => window.location.hash || defaultHash);
+
+	const hashChangeHandler = useCallback(() => {
```

Comment:
I think we can move this within the `useEffect`

Reply:
And no need to use the hook, it is only used within `useEffect` on mount/dismount